### PR TITLE
better attribute_name format checking

### PR DESCRIPTION
**Pull request by graft**
A tiny but crucial one - if you happen to add an attribute where attribute_name contains a space, Magma will approve the update, make the column, update the attributes table, then fail to come up after reboot because it can't find a column named "your_attribute_name ". The amendment requires going into postgres to update the attributes table, egad.

Here this is repaired with some more stringent format-checking on attribute names.

### CODE REVIEW

**Comment by coleshaw** [comment]
Oh this is great. I will take a closer look later. I found several other name validations too, that can break Magma, that I will include with the create project stuff.

File: magma/lib/magma/attribute.rb [hunk 1]
```diff
@@ -196,8 +196,8 @@ def validate_validation_json
     end
 
     def validate_attribute_name_format
-      return if attribute_name == attribute_name&.snake_case
-      errors.add(:attribute_name, "must be snake_case")
+      return if attribute_name =~ /\A[a-z][a-z0-9]*(_[a-z0-9]+)*\Z/
+      errors.add(:attribute_name, "must be snake_case with no spaces")
```
Comment (coleshaw) [inline magma/lib/magma/attribute.rb:200]: nit: should the error message clarify that it should also not start with a number?

**Review by coleshaw — APPROVED** [review]
looks good to me.

**Comment by coleshaw** [comment]
@graft , do you have additional updates for this PR?

File: magma/lib/magma/attribute.rb (hunk 1 above)
Comment (graft) [reply magma/lib/magma/attribute.rb:200]: Hmm, i might take that as implied by "snake_case".

**Pull request merged**